fxa/__main__.py

```diff
@@ -0,0 +1,205 @@
+                                       % auth[0])
+    elif create:
+        # Create a new user
+        if verbose:
+            print('# Creating the account...', end='', file=sys.stderr)
+            sys.stderr.flush()
+
+        try:
+            email, password = create_new_fxa_account(
+                os.getenv('FXA_USER_SALT', args.get('fxa_user_salt')),
+                args['account_server_url'], args['prefix'])
+        except (ClientError, ValueError) as e:
+            print('ERROR:\t %s' % e, file=sys.stderr)
+            sys.exit(1)
+
+        if verbose:
+            print("\b\b\b\t [OK]", file=sys.stderr)
+
+    if args['bearer']:
+        # Generate a Bearer Token for the user and write it into a file.
+        scopes = [s.strip() for s in re.split(';|,|\t|\n', args['scopes'])
+                  if s.strip()]
+        client_id = args['client_id']
+
+        if verbose:
+            print('# Generating the Bearer Token...', end='', file=sys.stderr)
+            sys.stderr.flush()
+
+        try:
+            token = get_bearer_token(email, password, scopes,
+                                     account_server_url,
+                                     oauth_server_url, client_id)
+        except ClientError as e:
+            print('ERROR:\t %s' % e, file=sys.stderr)
+            sys.exit(1)
+
+        if verbose:
+            print("\b\b\b\t [OK]", file=sys.stderr)
+
+        print('# ---- BEARER TOKEN INFO ----', file=fd)
+        print('# User: %s' % email, file=fd)
+        print('# Scopes: %s' % ' '.join(scopes), file=fd)
+        print('# Account: %s' % account_server_url, file=fd)
+        print('# Oauth: %s' % oauth_server_url, file=fd)
+        print('# Client ID: %s' % client_id, file=fd)
+        print('# ---------------------------', file=fd)
+        print('export OAUTH_BEARER_TOKEN="%s"\n' % token, file=fd)
+
+    if args['browserid']:
+        # Generate a BrowserID assertion for the user and write it into a file.
+        audience = args['audience']
+        duration = int(args['duration'])
+
+        if verbose:
+            print('# Creating the token...', end='', file=sys.stderr)
+            sys.stderr.flush()
+
+        try:
+            bid_assertion, client_state = get_browserid_assertion(
+                email, password, audience, account_server_url, duration)
+        except ClientError as e:
+            print('ERROR:\t %s' % e, file=sys.stderr)
+            sys.exit(1)
+
+        if verbose:
+            print("\b\b\b\t [OK]", file=sys.stderr)
+
+        print('# ---- BROWSER ID ASSERTION INFO ----', file=fd)
+        print('# User: %s' % email, file=fd)
+        print('# Audience: %s' % audience, file=fd)
+        print('# Account: %s' % account_server_url, file=fd)
+        print('# ------------------------------------', file=fd)
+        print('export FXA_BROWSERID_ASSERTION="%s"' % bid_assertion, file=fd)
+        print('export FXA_CLIENT_STATE="%s"\n' % client_state, file=fd)
+
+    fd.close()
```

Comment:
This will close stdout in the default close, which IMHO is slightly weird from within a function that may wind up being called by something else.